pkg/system/phase2_creating.go

```diff
@@ -143,6 +143,10 @@ func (r *Reconciler) ReconcilePhaseCreatingForMainClusters() error {
 		return err
 	}
 
+	if err := r.ReconcileObject(r.CoreAppConfig, r.SetDesiredCoreAppConfig); err != nil {
```

Comment:
maybe better to move that to the beginning since the core pod will not start before the configmap is created

Reply:
actually, I'm not sure you want to reconcile the configmap, since you will override any changes done by the user. you just need to create once it if it's not present

Reply:
not sure I understand what you mean, why I will override user changes if I'm reconciling the config map, we call `controllerutil.CreateOrUpdate` in `reconcileObject`, moreover, I dont see any user changes to the cm being deleted in the current behaviour.
I do accept moving the cm reconcile call to be called before the core pod.

Reply:
now when I think about it again, I was mistaken before. you do want to reconcile, just not to override any existing values. 
the reason we need to reconcile, is to handle any additional envs that we will add to the config map in future versions.

Reply:
@dannyzaken pushed a change regarding your last comment, let me know if the fix I implemented answers on the requirement.